tasks/Task.cpp

```diff
@@ -348,31 +379,40 @@ void Task::acquireFrame()
         throw runtime_error(ge.what());
     }
 
-    if (frame.image->IsIncomplete()) {
+    processFrame(frame.image);
+}
+
+void Task::processFrame(Arena::IImage* image)
+{
+    if (image->IsIncomplete()) {
         LOG_ERROR_S << "Image is not complete!! ";
         m_incomplete_images_count++;
         m_incomplete_images_sum++;
         return;
     }
 
-    size_t size = frame.image->GetSizeFilled();
-    size_t width = frame.image->GetWidth();
-    size_t height = frame.image->GetHeight();
-    PfncFormat pixel_format = static_cast<PfncFormat>(frame.image->GetPixelFormat());
+    size_t size = image->GetSizeFilled();
+    size_t width = image->GetWidth();
+    size_t height = image->GetHeight();
+    PfncFormat pixel_format = static_cast<PfncFormat>(image->GetPixelFormat());
     uint8_t data_depth = 0U;
     auto format = convertPixelFormatToFrameMode(pixel_format, data_depth);
 
-    Frame* out_frame = m_frame.write_access();
+    std::unique_ptr<Frame> out_frame(m_frame.try_write_access());
+    if (!out_frame) {
+        out_frame.reset(new base::samples::frame::Frame());
+    }
+
     if (width != out_frame->getWidth() || height != out_frame->getHeight() ||
         format != out_frame->getFrameMode()) {
-        out_frame->init(width, height, data_depth, format, 0, size);
+        out_frame->init(width, height, data_depth, format, -1, size);
```

Comment:
what does this change do?

Reply:
This -1 does not Fill the frame with data. Just creates it.